src/json.ts

```diff
@@ -19,17 +18,65 @@ import {
   union,
   unknown,
   Struct,
-} from 'superstruct';
-import type {
-  ObjectSchema,
-  Optionalize,
-  Simplify,
-} from 'superstruct/dist/utils';
+} from '@metamask/superstruct';
+import type { Context, Infer } from '@metamask/superstruct';
 
 import type { AssertionErrorConstructor } from './assert';
 import { assertStruct } from './assert';
 import { hasProperty } from './misc';
 
+/**
+ * Infer a type from an object struct schema.
+ *
+ * Copied from `superstruct`.
+ */
+type ObjectSchema = Record<string, Struct<any, any>>;
```

Comment:
`ObjectSchema`, `Optionalize`, and `Simplify` were added to the exports for `@metamask/superstruct` here: https://github.com/MetaMask/superstruct/pull/17. Can we use those instead of copying them?

Reply:
Yes we can! Fixed: https://github.com/MetaMask/utils/pull/185/commits/b13ee9c118d03db7286417c12179d5d054fbf099